wa/commands/create.py

```diff
@@ -35,6 +40,138 @@
 TEMPLATES_DIR = os.path.join(os.path.dirname(__file__), 'templates')
 
 
+class CreateDatabaseSubcommand(SubCommand):
+
+    name = 'database'
+    description = """
+    Create a Postgresql database which is compatible with the WA Postgres
+    output processor.
+    """
+
+    schemafilepath = 'postgres_schema.sql'
+
+    def initialize(self, context):
+        self.parser.add_argument('-a', '--host', default='localhost',
+                                 help='The host on which to create the database.')
+        self.parser.add_argument('-k', '--port', default='5432',
+                                 help='The port on which the PostgreSQL server is running.')
+        self.parser.add_argument('-u', '--username', default='postgres',
+                                 help='The username with which to connect to the server.')
+        self.parser.add_argument('-p', '--password',
+                                 help='The password for the user account.')
+        self.parser.add_argument('-d', '--dbname', default='wa',
+                                 help='The name of the database to create.')
+        self.parser.add_argument('-f', '--force', action='store_true',
+                                 help='Force overwrite the existing database if one exists.')
+        self.parser.add_argument('-m', '--force-update-config', action='store_true',
+                                 help='Force update the config file if an entry exists.')
+        self.parser.add_argument('-r', '--config-file', default=settings.user_config_file,
+                                 help='Path to the config file to be updated.')
+        self.parser.add_argument('-x', '--schema-version', action='store_true',
+                                 help='Display the current schema version.')
+
+    def execute(self, state, args):
+        self.get_schema(self.schemafilepath)
+        # Display the version if needed and exit
+        if args.schema_version:
+            self.logger.info(
+                'The current version is {}'.format(self.schemaversion))
+            sys.exit()
+        if not args.password:
+            self.parser.error(
+                'Please supply a password for the account used to connect.')
```

Comment:
I've made the password parameter optional, and the database connection will be attempted without one if a password is not given. It will notify the user if the connection is rejected.